compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4370,6 +4370,67 @@ private STNode createModuleVarDeclaration(STNode metadata, STNode publicQualifie
                 typedBindingPattern, assign, expr, semicolon);
     }
 
+    private STNode createSimpleVarDeclaration(STNode metadata) {
+        STNode publicQualifier = STNodeFactory.createEmptyNode();
+        return createSimpleVarDeclaration(metadata, publicQualifier, new ArrayList<>(), true);
+    }
+
+    private STNode createSimpleVarDeclaration(STNode metadata, STNode publicQualifier, List<STNode> qualifiers,
+                                              boolean isModuleVar) {
+        STNode emptyNode = STNodeFactory.createEmptyNode();
+        STNode simpleTypeDescIdentifier = SyntaxErrors.createMissingTokenWithDiagnostics(SyntaxKind.IDENTIFIER_TOKEN,
+                DiagnosticErrorCode.ERROR_MISSING_TYPE_DESC);
+        STNode identifier = SyntaxErrors.createMissingTokenWithDiagnostics(SyntaxKind.IDENTIFIER_TOKEN,
+                DiagnosticErrorCode.ERROR_MISSING_IDENTIFIER);
+        STNode simpleNameRef = STNodeFactory.createSimpleNameReferenceNode(simpleTypeDescIdentifier);
+        STNode semicolon = SyntaxErrors.createMissingTokenWithDiagnostics(SyntaxKind.SEMICOLON_TOKEN,
+                DiagnosticErrorCode.ERROR_MISSING_SEMICOLON_TOKEN);
```

Comment:
This part is a duplicate of 4409 to 4416. It simply cannot extract to a method though.